frappe/core/doctype/doctype/doctype.py

```diff
@@ -1192,13 +1188,20 @@ def check_illegal_default(d):
 					_("Default value for {0} must be in the list of options.").format(frappe.bold(d.fieldname))
 				)
 
-	def check_precision(d):
-		if (
-			d.fieldtype in ("Currency", "Float", "Percent")
-			and d.precision is not None
-			and not (1 <= cint(d.precision) <= 6)
-		):
-			frappe.throw(_("Precision should be between 1 and 6"))
+		if d.default and d.fieldtype == "Int":
+			if not isinstance(d.default, int):
```

Comment:
does this work? default field is a text field - not sure if it translates to int implicitly